src/main/webapp/forms/counter.jsp

```diff
@@ -1,63 +1,39 @@
-﻿<%@ include file="../configuration.jsp" %>
+<%@ page import="de.uni_tuebingen.ub.nppm.db.*" isThreadSafe="false" %>
 
-<%@ page import="java.sql.Connection" isThreadSafe="false" %>
-<%@ page import="java.sql.DriverManager" isThreadSafe="false" %>
-<%@ page import="java.sql.ResultSet" isThreadSafe="false" %>
-<%@ page import="java.sql.SQLException" isThreadSafe="false" %>
-<%@ page import="java.sql.Statement" isThreadSafe="false" %>
+﻿<%@ include file="../configuration.jsp" %>
 
 <%
-  int akt = 0;
-  int max = 0;
-
-  int id = -1;
-  String title = request.getParameter("title");
-  String guest = "";
-  if (title.contains("gast_")) {
-    title = title.substring(5);
-    guest = "gast_";
-  }
-  int filter = 0;
-  String filterParameter = request.getParameter("filterParameter")==null?"":request.getParameter("filterParameter");
-  try {
-    id = Integer.parseInt(request.getParameter("ID"));
-    filter = Integer.parseInt(request.getParameter("filter"));
-  } catch (NumberFormatException e) {}
-
-
-  Connection cn = null;
-  Statement st = null;
-  ResultSet rs = null;
-
-  try {
-    String sql = "SELECT * FROM "+title;
-
-    Class.forName( sqlDriver );
-    cn = DriverManager.getConnection( sqlURL, sqlUser, sqlPassword );
-    st = cn.createStatement();
-
-    rs = st.executeQuery("SELECT * FROM datenbank_filter WHERE Formular='"+guest+title+"' AND Nummer='"+filter+"';");
-    if (rs.next()) {
-      sql = rs.getString("SQLString");
+    int akt = 0;
+    int max = 0;
+
+    int id = -1;
+    String title = request.getParameter("title");
+    String guest = "";
+    if (title.contains("gast_")) {
+        title = title.substring(5);
+        guest = "gast_";
+    }
+    int filter = 0;
+    String filterParameter = request.getParameter("filterParameter") == null ? "" : request.getParameter("filterParameter");
+    try {
+        id = Integer.parseInt(request.getParameter("ID"));
+        filter = Integer.parseInt(request.getParameter("filter"));
+    } catch (NumberFormatException e) {
     }
 
+    String sql = DatenbankDB.getFilterSql(guest + title, filter);
     sql = sql.replace("*", "count(*) c");
     sql = sql.replace("###", filterParameter);
 
-    rs = st.executeQuery(sql+(sql.contains("WHERE")?" AND ":" WHERE ")+title+".ID < "+id);
-    if (rs.next())
-      akt = rs.getInt("c") + 1;
+    Object[] columns = AbstractBase.getRowNative(sql + (sql.contains("WHERE") ? " AND " : " WHERE ") + title + ".ID < " + id);
```

Comment:
In my built of this branch i got an execption on line 28.

`Caused by: java.lang.ClassCastException: class java.math.BigInteger cannot be cast to class [Ljava.lang.Object; (java.math.BigInteger and [Ljava.lang.Object; are in module java.base of loader 'bootstrap')`

Reply:
solved